test/torchaudio_unittest/datasets/cmudict_test.py

```diff
@@ -0,0 +1,97 @@
+import os
+from pathlib import Path
+import string
+
+from torchaudio.datasets import CMUDict
+
+from torchaudio_unittest.common_utils import (
+    TempDirMixin,
+    TorchaudioTestCase,
+)
+
+
+def get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    mocked_dictionary = [
+        ";;; # CMUdict  --  Major Version: 0.07",
+        ";;; ",
+        ";;; # $HeadURL$",
+        "!EXCLAMATION-POINT  EH2 K S K L AH0 M EY1 SH AH0 N P OY2 N T",
+        "\"CLOSE-QUOTE  K L OW1 Z K W OW1 T",
+        "AALIYAH  AA2 L IY1 AA2",
+        "AALSETH  AA1 L S EH0 TH",
+        "AAMODT  AA1 M AH0 T",
+        "AANCOR  AA1 N K AO2 R",
+        "AARDEMA  AA0 R D EH1 M AH0",
+        "AARDVARK  AA1 R D V AA2 R K",
+    ]
+    mocked_symbols = [
+        "AA0", "AA1", "AA2", "AH0", "AO2", "D", "EH0", "EH2", "IY1", "K", "L", "M",
+        "N", "P", "OW1", "OY2", "R", "S", "SH", "T", "TH", "W", "V", "Z",
+    ]
+
+    dict_file = os.path.join(root_dir, "cmudict-0.7b")
+    symbol_file = os.path.join(root_dir, "cmudict-0.7b.symbols")
+
+    with open(dict_file, "w") as txt:
+        txt.write("\n".join(mocked_dictionary))
+
+    with open(symbol_file, "w") as txt:
+        txt.write("\n".join(mocked_symbols))
+
+    mocked_data = []
+    for ent in mocked_dictionary:
+        if len(ent) and ent[:3] == ";;;":
```

Comment:
In tests, generally speaking, it is better to avoid the same logic implemented in the library, and keep the test logic as simple as possible. (Note: we use tests to debug the library, but finding a logical error in the test code is much harder when the test simply reports "pass".)

1. Splitting the `mocked_dictionary` into header, punctuations, and non-punctuations, the preparation logic becomes cleaner.
2. It is ideal to avoid parsing the returned `mocked_data` for exclusion of punctuation.

So I suggest to change the helper function to support exclusion of punctuations, making it the responsibility of helper function to return the proper mock data.

```python
def get_mock_dataset(root_dir, return_punc=False):
    header = [
        ";;; # CMUdict  --  Major Version: 0.07",
        ...
    ]

    puncs = [
        "!EXCLAMATION-POINT EH2 K S K L AH0 M EY1 SH AH0 N P OY2 N T",
        ...
    ]

    words = [
        "AALIYAH  AA2 L IY1 AA2",
        ...
    ]
    ...
    with open(dict_file, "w") as fileobj:
        for section in [header, punch, words]:
            for line in section:
                fileobj.write(line)
                fileobj.write("\n")
    ...

    data_to_include = puncs + words if return_punc else words
    mocked_data = []
    for ent in data_to_include:
        word, phones = ent.split("  ")
        mocked_data.append((word, phones.split(" ")))
    return mocked_data
```

then in the test class, we can generate two mock datasets, one with punctuation and one without punctuations, and use the different `root_dir` value for different test.
(also, the use of `if dataset.exclude_punctuations` should be avoided in test logic to keep it simple.)

Reply:
Thanks for the suggestion! I've rewritten the testing part with this logic [here](https://github.com/pytorch/audio/pull/1627/files#diff-017895ceaff7f853256bf11da184c3d6873e5322b9ee3c0051208639aecaa3d2R12).